content/blog/2018-08-05-what-is-dx-developer-experience/index.md

```diff
@@ -0,0 +1,169 @@
+
+![d3-js documentation transitions](./d3-js-documentation-transitions.png)
+
+I specifically choose this section from the official D3JS page, as it has elements that API document authors should avoid — not disparaging the D3 library here, highly respect the work Mike Bostrom the creator is doing.
+
+* Context unclear to beginners — Transitions is given as the title, not once is ever given a formal definition
+
+* Esoteric Vocabulary — within the paragraphs, it mentions “transformations”, “easing functions” and “D3 Interpolators” which already throw developers who have never worked with CSS transitions and D3 straight into the deep end.
+
+* Unclear Connections — There is no clear link between D3 Interpolators and Transitions
+
+Reading over the section again, the author could have dramatically improved the readers understanding, just by simply attaching hyperlinks to the page itself.
+
+## Train Developers to Overcome the Barriers
+VueJS’s tutorials are an excellent example, take a look at the amount of support for absolute beginners.
+
+VueJS documents on Declarative Rendering
+The Vue team even went a step further with their interactive lessons using scrimba
+
+![Vue-js declaration rendering tutorial using Scrimba](./vue-js-declarative-rendering.png)
+
+### Feedback Quality
+
+Whenever something goes wrong, it’s essential that developers receive an unambiguous, detailed response of the error, a trace-stack and even a way to deal with the problem. This may all seem obvious, however not many tools out either are very specific with the issue at hand, meaning that developers spend time searching for problems they have presumably caused, time that could be spent developing
+
+![Beautified React error message on component exports](./react-error-message-on-js-exports.png)
+
+The error message not only states the error, but it also gives the developer the likely solution to the problem *per se*. Let’s compare this to an example of the error messages of Node v4.
+
+![Set timeout Node v4 error message](./set-timeout-node-v4-error-message.png)
+
+I want to thank Anna Henningsen for her talk on Node.js at JSConf Iceland 2018. She mentions how this error message doesn’t really describe the problem that exists.
+
+With Anna’s help, the change to the error message for Node v5 is the one made below.
+
+![Set timeout Node v5 error message](./set-timeout-node-v5-error-message.png)
+
+Still further improvement can be made; to state that the callback is the second argument of the setTimeout method.
+
+### Feedback Speed
+Reducing the time that it takes for code, to affect the changes made in a product, gives developers a sense of tactile instant feedback — providing developers do not have to wait long. Enables them to expend less effort assessing the small micro changes they make to the product. Below shows ReactJS ‘Hot Module Reloading’ feature, which reloads the browser whenever there is a change made to the UI.
```

Comment:
Change to "This enables them to expend"

Reply:
Could you be a bit more specific - note down the line or phrase that needs to be changed

Reply:
This needs more explanation.

Could you zone in the exact words that needs to replaced?

Oh and did you mean "This enables them to expand", rather than "This enables them to expend"?

Reply:
Okay so instead of  "Enables them to expend less effort assessing the small micro changes they make to the product." it should be "This enables them to expend less effort assessing the small micro changes they make to the product."

Reply:
Cool I'll make that change